sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Rules/Hotspots/DoNotHardcodeCredentials.cs

```diff
@@ -1,85 +1,108 @@
-{
-    [DiagnosticAnalyzer(LanguageNames.CSharp)]
-    [Rule(DiagnosticId)]
-    public sealed class DoNotHardcodeCredentials : DoNotHardcodeCredentialsBase
-    {
-        private static readonly DiagnosticDescriptor rule =
-            DiagnosticDescriptorBuilder.GetDescriptor(DiagnosticId, MessageFormat, RspecStrings.ResourceManager,
-                isEnabledByDefault: false);
-
-        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics { get; } =
-            ImmutableArray.Create(rule);
-
-        protected override void Initialize(ParameterLoadingAnalysisContext context)
-        {
-            context.RegisterSyntaxNodeActionInNonGenerated(
-                new VariableDeclarationBannedWordsFinder(this).GetAnalysisAction(rule),
-                SyntaxKind.VariableDeclarator);
-
-            context.RegisterSyntaxNodeActionInNonGenerated(
-                new AssignmentExpressionBannedWordsFinder(this).GetAnalysisAction(rule),
-                SyntaxKind.SimpleAssignmentExpression);
-        }
-
-        private class VariableDeclarationBannedWordsFinder : CredentialWordsFinderBase<VariableDeclaratorSyntax>
-        {
-            public VariableDeclarationBannedWordsFinder(DoNotHardcodeCredentialsBase analyzer) : base(analyzer) { }
-
-            protected override string GetAssignedValue(VariableDeclaratorSyntax syntaxNode) =>
-                syntaxNode.Initializer?.Value.GetStringValue();
-
-            protected override string GetVariableName(VariableDeclaratorSyntax syntaxNode) =>
-                syntaxNode.Identifier.ValueText;
-
-            protected override bool IsAssignedWithStringLiteral(VariableDeclaratorSyntax syntaxNode, SemanticModel semanticModel) =>
-                (syntaxNode.Initializer?.Value is LiteralExpressionSyntax literalExpression) &&
-                literalExpression.IsKind(SyntaxKind.StringLiteralExpression) &&
-                syntaxNode.IsDeclarationKnownType(KnownType.System_String, semanticModel);
-        }
-
-        private class AssignmentExpressionBannedWordsFinder : CredentialWordsFinderBase<AssignmentExpressionSyntax>
-        {
-            public AssignmentExpressionBannedWordsFinder(DoNotHardcodeCredentialsBase analyzer) : base(analyzer) { }
-
-            protected override string GetAssignedValue(AssignmentExpressionSyntax syntaxNode) =>
-                syntaxNode.Right.GetStringValue();
-
-            protected override string GetVariableName(AssignmentExpressionSyntax syntaxNode) =>
-                (syntaxNode.Left as IdentifierNameSyntax)?.Identifier.ValueText;
-
-            protected override bool IsAssignedWithStringLiteral(AssignmentExpressionSyntax syntaxNode, SemanticModel semanticModel) =>
-                syntaxNode.IsKind(SyntaxKind.SimpleAssignmentExpression) &&
-                syntaxNode.Left.IsKnownType(KnownType.System_String, semanticModel) &&
-                syntaxNode.Right.IsKind(SyntaxKind.StringLiteralExpression);
-        }
-    }
-}
+﻿/*
```

Comment:
line endings changed, could you revert please? (I cannot tell if there's a change or just rename)

(it is a nuissance, we'll have to adopt some common practice in the dotnet area and document it and enforce it with tooling, because currently the codebase is split evenly between unix and windows line endings...)

Reply:
or maybe just github cannot tell the diff.. I now look at the first commit

Reply:
I tried changing line ending, but no matter what I put as line ending, github still marks the whole file as changed unfortunately